services/src/main/java/org/keycloak/authentication/authenticators/client/JWTClientValidator.java

```diff
@@ -192,12 +194,50 @@ public void validateToken() {
         }
     }
 
+    private long calculateLifespanInSeconds()  {
+        if ((token.getExp() == null || token.getExp() <= 0) && (token.getIat() == null || token.getIat() <= 0)) {
+            throw new RuntimeException("Token cannot be validated. Neither the exp nor the iat claim are present.");
+        }
+
+        // rfc-9068 marks exp and iat claims are required: https://datatracker.ietf.org/doc/html/rfc9068#section-2.2
```

Comment:
This comment is a little bit confusing IMO as it references RFC9068. But RFC9068 is related to "JWT access tokens", but the JWT used for client authentication may not necessarily be an "access token" per my understanding and hence it should rather rely on the other specs https://datatracker.ietf.org/doc/html/rfc7523#section-3 , which mentions `iat` as optional. WDYT?

The code actually allows `iat` as optional and you even have test for it. So the behaviour of this PR is OK IMO . Just this single comment may need some update, but this is not a blocker for this PR IMO.

Reply:
@mposolda That section says:

> In order to issue an access token response as described in OAuth 2.0
   RFC6749 or to rely on a JWT for client authentication, the
   authorization server MUST validate the JWT according to the criteria
   below.

So the part that says `or to rely on a JWT for client authentication` applies here... Or am I missing something?

Reply:
Ah! I copied the other RFC, right I understand, doing the change now.

Reply:
Done!

Reply:
@rmartinc Thanks! I hope to merge the PR once tests are OK